Firestore/core/src/firebase/firestore/core/transaction_runner.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_CORE_TRANSACTION_RUNNER_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_CORE_TRANSACTION_RUNNER_H_
+
+#include <memory>
+
+#include "Firestore/core/src/firebase/firestore/core/transaction.h"
+#include "Firestore/core/src/firebase/firestore/remote/exponential_backoff.h"
+#include "Firestore/core/src/firebase/firestore/remote/remote_store.h"
+#include "Firestore/core/src/firebase/firestore/util/async_queue.h"
+#include "Firestore/core/src/firebase/firestore/util/status.h"
+
+namespace core = firebase::firestore::core;
+namespace util = firebase::firestore::util;
+namespace remote = firebase::firestore::remote;
+
+namespace firebase {
```

Comment:
For Obj-C++ code (which this is since the implementation is in a .h file), we typically wrap the body in

```
NS_ASSUME_NONNULL_BEGIN
...
NS_ASSUME_NONNULL_END
```

This automatically interprets any references to Obj-C objects (ex. `FSTFoo*`) as nonnullable (ex: `FSTFoo*_Nonnull`) which makes our code safer.

Reply:
My bad. 😄